pkg/rc/replication_controller.go

```diff
@@ -0,0 +1,40 @@
+package rc
+
+type ReplicationController interface {
+	// It is expected that the below functions will be used by the orchestrator,
+	// in response to user-requested deploy events.
+	// They affect the desired behavior of the replication controller.
+
+	// IncreaseReplicas(n) increases the number of desired replicas by n.
+	// This returns the new desired number if successful, or the old number and an error.
+	// It is expected that the replication controller will attempt to schedule its pod onto a node returned by EligibleNodes().
+	IncreaseReplicas(n int) (int, error)
+
+	// DecreaseReplicas(n) decreases the number of desired replicas by n.
+	// This returns the new desired number if successful, or the old number and an error.
+	DecreaseReplicas(n int) (int, error)
+
+	// Disable() causes the replication controller to stop reacting to changes in the number of desired replicas.
+	Disable() error
+
+	// It is expected that the below functions are used by the replication controller
+	// to determine whether observed state matches desired state.
+
+	// EligibleNodes() returns the nodes this replication controller could possibly run on.
+	EligibleNodes() ([]string, error)
+
+	// CurrentNodes() returns the nodes this replication controller is currently running on.
+	CurrentNodes() ([]string, error)
+
+	// DesiredReplicas() returns the number of desired replicas for this replication controller.
+	DesiredReplicas() (int, error)
+
+	// It is expected that the below functions are used by the replication controller
+	// to bring observed state in line with desired state.
+
+	// Schedule(node) schedules the pod managed by this replication controller onto the specified node
+	Schedule(node string) error
```

Comment:
that's true but I'd expect replication controllers to only call this with a node that is in the `Eligible - Current` set. It may be the case that this will be non-exported. I'll see as I continue to write more implementation code.